### PR TITLE
feat: replaced `semver` dependency with custom implementation

### LINKED ISSUE #1593: Remove `semver` dependency
This issue is based on the discussion #1592.

### PR DESCRIPTION
**Pull request by nedsalk**
This PR replaces the `semver` dependency with an ad hoc semver implementation that satisfies the needs of the utilities that were depending on `semver`. I didn't write any tests for it because those utilities are covered with tests; if there was something wrong with the implementation, those tests would've failed.

This change reduces the final bundle size by 19.86 kB.

Closes #1593.

### CODE REVIEW

File: packages/versions/src/lib/semver.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    const semver2 = new Semver(version2);
+    return semver1.minor === semver2.minor;
+  }
+
+  public static patchEq(version1: string, version2: string): boolean {
+    const semver1 = new Semver(version1);
+    const semver2 = new Semver(version2);
+    return semver1.patch === semver2.patch;
+  }
+}
```
Comment (arboleya) [inline packages/versions/src/lib/semver.ts:51]: Looking good! Two only considerations so far:
 1. The use of classes seems unnecessary — can we use simple functions?
 1. Consider adding a `semver.test.ts` covering 💯 everything

> _Please apply the same mentality to all other similar PRs around tree-shakeability._

```ts
// ALERT: pseudo-code
// Just playing around with your stuff :)

export const parseVersionSegments = (version: string) => {
  const [major, minor, patch] = version.split('.').map((v) => parseInt(v, 10));
  return { major, minor, patch };
};

export type VersionSegment = 'major' | 'minor' | 'patch';

export const isSegmentEq = (v1: string, v2: string, segment: VersionSegment) => {
  const seg1 = parseVersionSegments(v1)[segment];
  const seg2 = parseVersionSegments(v2)[segment];
  return seg1 === seg2;
};

export const diffVersions = (v1: string, v2: string) => {
  const segsA = parseVersionSegments(v1);
  const segsB = parseVersionSegments(v2);
  if (segsA.major !== segsB.major) {
    return segsA.major - segsB.major;
  }
  if (segsA.minor !== segsB.minor) {
    return segsA.minor - segsB.minor;
  }
  return segsA.patch - segsB.patch;
};

export const isEq = (v1: string, v2: string) => diffVersions(v1, v2) === 0;
export const isGt = (v1: string, v2: string) => diffVersions(v1, v2) > 0;

export const isMajorEq = (v1: string, v2: string) => isSegmentEq(v1, v2, 'major');
export const isMinorEq = (v1: string, v2: string) => isSegmentEq(v1, v2, 'minor');
export const isPatchEq = (v1: string, v2: string) => isSegmentEq(v1, v2, 'patch');

```

Comment (nedsalk) [reply packages/versions/src/lib/semver.ts]: https://github.com/FuelLabs/fuels-ts/pull/1594/commits/29b918957c19b37ce026e1dbfca69b9515a50441

Comment (nedsalk) [reply packages/versions/src/lib/semver.ts]: https://github.com/FuelLabs/fuels-ts/pull/1594/commits/4a9f53b519087674ef92b32f3f84e6bb917d10ea

**Review by danielbate — APPROVED** [review]
:shipit: Just needs an empty changeset.

File: .changeset/beauty-die-joy.md [hunk 2]
```diff
@@ -0,0 +1 @@
+TODO: delete me
```
Comment (danielbate) [inline .changeset/beauty-die-joy.md:1]:
```suggestion
---
---

```

Comment (arboleya) [reply .changeset/beauty-die-joy.md:1]: This came from the precociously-deleted `rc: beetle`.

This file can be removed entirely.

Comment (nedsalk) [reply .changeset/beauty-die-joy.md:1]: I'll let this beauty die joyfully, then.

**Review by danielbate — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**